message/message.go

```diff
@@ -131,22 +132,39 @@ func (m *Message) AddArg(arg string) *Message {
 	return m
 }
 
-// AsSlice outputs the given Message as a string slice.
-// The slice contains the string form of the Message's word in index 0, and
-// the arguments as index 1 upwards, if any.
-func (m *Message) AsSlice() []string {
-	slice := []string{m.word.String()}
-	for _, arg := range m.args {
-		slice = append(slice, arg)
-	}
-	return slice
+func escapeArgument(input string) (output string) {
+	return "'" + strings.Replace(input, "'", `'\''`, -1) + "'"
 }
 
 // Pack outputs the given Message as raw bytes representing a BAPS3 message.
 // These bytes can be sent down a TCP connection to a BAPS3 server, providing
 // they are terminated using a line-feed character.
-func (m *Message) Pack() ([]byte, error) {
-	return Pack(m.word.String(), m.args)
+func (m *Message) Pack() (packed []byte, err error) {
+	output := new(bytes.Buffer)
+
+	_, err = output.WriteString(m.word.String())
+	if err != nil {
+		return
+	}
+
+	for _, a := range m.args {
+		// Escape arg if needed
+		for _, c := range a {
+			if c < unicode.MaxASCII && (unicode.IsSpace(c) || strings.ContainsRune(`'"\`, c)) {
+				a = escapeArgument(a)
+				break
+			}
+		}
+
+		_, err = output.WriteString(" " + a)
+		if err != nil {
+			return
+		}
+	}
+	output.WriteString("\n")
+
+	packed = output.Bytes()
+	return
```

Comment:
Good call I think, but if you're doing this then it might be time to make the tokeniser return `[]Message`.  This depends on solving the word-as-string-or-enum question, though